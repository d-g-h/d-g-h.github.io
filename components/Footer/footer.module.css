.footer {
  display: block;

  @media (min-width: 640px) {
    display: flex;
    justify-content: space-between;
  }
}

.title {
  text-decoration: underline;
}

.skill {
  font-size: 0.75rem;
}

.education {
  @media (min-width: 640px) {
    margin-left: 0.25rem;
    text-align: right;
  }
}
.ast {
  @media print {
    color: white;
    font-size: 0.25rem;
    -webkit-print-color-adjust: exact;
  }

  font-size: 0;
}

@media (min-width: 640px) {
  .skills {
    max-width: 65%;
  }
}
