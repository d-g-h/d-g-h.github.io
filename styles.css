html{font-family:sans-serif;-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%}body{margin:0}article,aside,details,figcaption,figure,footer,header,hgroup,main,menu,nav,section,summary{display:block}audio,canvas,progress,video{display:inline-block;vertical-align:baseline}audio:not([controls]){display:none;height:0}[hidden],template{display:none}a{background-color:transparent}a:active,a:hover{outline:0}abbr[title]{border-bottom:1px dotted}b,strong{font-weight:700}dfn{font-style:italic}h1{font-size:2em;margin:.67em 0}mark{background:#ff0;color:#000}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sup{top:-.5em}sub{bottom:-.25em}img{border:0}svg:not(:root){overflow:hidden}figure{margin:1em 40px}hr{box-sizing:content-box;height:0}pre{overflow:auto}code,kbd,pre,samp{font-family:monospace,monospace;font-size:1em}button,input,optgroup,select,textarea{color:inherit;font:inherit;margin:0}button{overflow:visible}button,select{text-transform:none}button,html input[type=button],input[type=reset],input[type=submit]{-webkit-appearance:button;cursor:pointer}button[disabled],html input[disabled]{cursor:default}button::-moz-focus-inner,input::-moz-focus-inner{border:0;padding:0}input{line-height:normal}input[type=checkbox],input[type=radio]{box-sizing:border-box;padding:0}input[type=number]::-webkit-inner-spin-button,input[type=number]::-webkit-outer-spin-button{height:auto}input[type=search]{-webkit-appearance:textfield;box-sizing:content-box}input[type=search]::-webkit-search-cancel-button,input[type=search]::-webkit-search-decoration{-webkit-appearance:none}fieldset{border:1px solid silver;margin:0 2px;padding:.35em .625em .75em}legend{border:0;padding:0}textarea{overflow:auto}optgroup{font-weight:700}table{border-collapse:collapse;border-spacing:0}td,th{padding:0}.hljs{display:block;overflow-x:auto;padding:.5em;background:#00222b;color:#839496;-webkit-text-size-adjust:none}.diff .hljs-header,.hljs-comment,.hljs-doctype,.hljs-pi,.lisp .hljs-string{color:#aaa}.css .hljs-tag,.hljs-addition,.hljs-keyword,.hljs-request,.hljs-status,.hljs-winutils,.method,.nginx .hljs-title{color:#859900}.hljs-command,.hljs-doctag,.hljs-hexcolor,.hljs-link_url,.hljs-number,.hljs-regexp,.hljs-rule .hljs-value,.hljs-string,.hljs-tag .hljs-value,.tex .hljs-formula{color:#2aa198}.css .hljs-function,.hljs-built_in,.hljs-chunk,.hljs-decorator,.hljs-id,.hljs-identifier,.hljs-localvars,.hljs-name,.hljs-title,.vhdl .hljs-literal{color:#268bd2}.hljs-attribute,.hljs-class .hljs-title,.hljs-constant,.hljs-link_reference,.hljs-parent,.hljs-type,.hljs-variable,.lisp .hljs-body,.smalltalk .hljs-number{color:#b58900}.css .hljs-pseudo,.diff .hljs-change,.hljs-attr_selector,.hljs-cdata,.hljs-header,.hljs-pragma,.hljs-preprocessor,.hljs-preprocessor .hljs-keyword,.hljs-shebang,.hljs-special,.hljs-subst,.hljs-symbol,.hljs-symbol .hljs-string{color:#cb4b16}.hljs-deletion,.hljs-important{color:#dc322f}.hljs-link_label{color:#6c71c4}.tex .hljs-formula{background:#073642}html{box-sizing:border-box}*,:after,:before{box-sizing:inherit}@font-face{font-family:icomoon;src:url(75cf10f313532a120f287de6a31e64bb.eot);src:url(75cf10f313532a120f287de6a31e64bb.eot?#iefix-rsav07) format("embedded-opentype"),url(5f49880e43263c57f009240776e181e0.woff) format("woff"),url(00958aa80c387f15ac23398ce58da5e6.ttf) format("truetype"),url(f3efa91a2cf7b68aee70aa70dc8eb3db.svg#icomoon) format("svg");font-weight:400;font-style:normal}.icon-file-text:before,.icon-mark-github:before{font-family:icomoon;speak:none;font-style:normal;font-weight:400;font-variant:normal;text-transform:none;line-height:1;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale}.icon-mark-github:before{content:"\E600"}.icon-file-text:before{content:"\E601"}body{color:#333;font:13px/1.4 system,-apple-system,\.SFNSDisplay-Regular,Helvetica Neue,Helvetica,arial,nimbussansl,liberationsans,freesans,clean,sans-serif}a{color:#4078c0;text-decoration:none}a:hover{text-decoration:underline}a span{font-size:8px;padding-left:.25em}h1{text-align:center}h2{margin-bottom:0}h3,h4,h5,h6{color:#666}li{list-style:none}p{margin:.5em 0}time{color:#6a737d;font:13px/1 system,-apple-system,\.SFNSDisplay-Regular,Helvetica Neue,Helvetica,arial,nimbussansl,liberationsans,freesans,clean,sans-serif}.logo{background-image:url("/assets/images/dgh.png");background-size:contain;border-radius:50%;display:block;height:3em;margin:0 auto;width:3em}header{padding-top:1rem}nav ul{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;margin:0;padding:0}nav ul li{-webkit-align-self:flex-end;-ms-flex-item-align:end;align-self:flex-end;padding:1em;text-align:center}nav ul a{font-size:12px}.contribute{font-size:10px}.icon-file-text:before,.icon-mark-github:before{font-size:24px;padding:4px}.container{border-top:1px solid #999;font-size:16px;margin:0 auto;max-width:800px;padding:.5em}video{display:block;margin:0 auto;max-width:100%}.contact-info,.job-title{color:#4078c0;text-align:center}.contact-info span{font-size:50%}.job-time{color:#999;font-size:10.66667px}.resume.container{border:0}.resume.container h2,.resume.container h3,.resume.container h4,.resume.container h5,.resume.container p{margin:0}.resume.container h2,.resume.container h3,.resume.container h4,.resume.container h5{color:#333}.resume.container p{color:#666}.resume.container p span{font-size:75%}.resume.container a{display:block}article{margin-top:1em}@media print{.resume{font-size:85%}}