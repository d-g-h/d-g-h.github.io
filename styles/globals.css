:root {
  --light-gray-color: #adbac7;
  --dark-gray-color: #22272e;
  --light-color: #ffffff;
  --blue-color: #539bf5;
  --lime-color: #8ddb8c;
}

html {
  font-family:
    -apple-system,
    BlinkMacSystemFont,
    Segoe UI,
    Roboto,
    Oxygen,
    Ubuntu,
    Cantarell,
    Fira Sans,
    Droid Sans,
    Helvetica Neue,
    sans-serif;
}

body {
  background-color: var(--dark-gray-color);
  color: var(--light-gray-color);
  padding: 0;
  margin: 0;
}

ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

a {
  color: var(--blue-color);
}

@media print {
  body,
  a {
    color: var(--dark-gray-color);
  }
}
