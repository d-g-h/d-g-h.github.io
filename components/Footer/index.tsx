type FooterProps = {
  skills?: Skill[];
  education: Education[];
};

type Skill = {
  key: string;
  name: string;
};

type Education = {
  key: string;
  name: string;
};

import styles from "@/components/Footer/footer.module.css";

export default function Footer({ skills, education }: Readonly<FooterProps>) {
  const ATS = "ATS";
  return (
    <div className={styles.footer}>
      {skills ? (
        <div className={styles.skills} style={{ textAlign: "left" }}>
          <h2 className={styles.title} style={{ textDecoration: "underline" }}>
            Skills
          </h2>

          <ul style={{ fontSize: "0.75rem" }}>
            {skills.map((skill: Skill) =>
              skill.key === ATS ? (
                <li key={skill.key} className={styles.ast}>
                  {skill.name}
                </li>
              ) : (
                <li key={skill.key}>{skill.name}</li>
              )
            )}
          </ul>
        </div>
      ) : (
        ""
      )}
      <div className={skills ? styles.education : ""}>
        <h2 className={styles.title}>Education</h2>
        <ul className="sub">
          {education.map((education: Education) => (
            <li key={education.key}>{education.name}</li>
          ))}
        </ul>
      </div>
    </div>
  );
}
