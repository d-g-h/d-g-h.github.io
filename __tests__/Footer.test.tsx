import { render, screen } from "@testing-library/react";
import Footer from "@/components/Footer";
import styles from "@/components/Footer/footer.module.css";

describe("Footer", () => {
  describe("When skills are provided", () => {
    const skills = [
      { key: "1", name: "JavaScript" },
      { key: "2", name: "React" },
    ];
    const education = [{ key: "1", name: "Bachelor's Degree" }];

    it("renders skills section", () => {
      render(<Footer skills={skills} education={education} />);
      expect(screen.getByText("Skills")).toBeInTheDocument();
      expect(screen.getByText("JavaScript")).toBeInTheDocument();
      expect(screen.getByText("React")).toBeInTheDocument();
    });

    it("renders ATS skill with special class", () => {
      const atsSkills = [{ key: "ATS", name: "ATS Skill" }];
      render(<Footer skills={atsSkills} education={education} />);
      const atsSkill = screen.getByText("ATS Skill");
      expect(atsSkill).toHaveClass(styles.ast);
    });

    it("education section has correct class", () => {
      render(<Footer skills={skills} education={education} />);
      const educationSection = screen.getByText("Education").closest("div");
      expect(educationSection).toHaveClass(styles.education);
    });

    it("matches snapshot when skills and education are provided", () => {
      const { container } = render(
        <Footer skills={skills} education={education} />
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("When skills are not provided", () => {
    const education = [
      { key: "1", name: "Bachelor's Degree" },
      { key: "2", name: "Master's Degree" },
    ];

    it("does not render skills section", () => {
      render(<Footer education={education} />);
      expect(screen.queryByText("Skills")).not.toBeInTheDocument();
    });

    it("renders education section correctly", () => {
      render(<Footer education={education} />);
      expect(screen.getByText("Education")).toBeInTheDocument();
      expect(screen.getByText("Bachelor's Degree")).toBeInTheDocument();
      expect(screen.getByText("Master's Degree")).toBeInTheDocument();
    });

    it("education section does not have special class", () => {
      render(<Footer education={education} />);
      const educationSection = screen.getByText("Education").closest("div");
      expect(educationSection).not.toHaveClass(styles.education);
    });

    it("matches snapshot when only education is provided", () => {
      const { container } = render(<Footer education={education} />);
      expect(container).toMatchSnapshot();
    });
  });
});
