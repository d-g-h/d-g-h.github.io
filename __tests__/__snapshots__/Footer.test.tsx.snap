// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Footer When skills are not provided matches snapshot when only education is provided 1`] = `
<div>
  <div
    class="footer"
  >
    <div
      class=""
    >
      <h2
        class="title"
      >
        Education
      </h2>
      <ul
        class="sub"
      >
        <li>
          Bachelor's Degree
        </li>
        <li>
          Master's Degree
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Footer When skills are provided matches snapshot when skills and education are provided 1`] = `
<div>
  <div
    class="footer"
  >
    <div
      class="skills"
      style="text-align: left;"
    >
      <h2
        class="title"
      >
        Skills
      </h2>
      <ul>
        <li
          class="skill"
        >
          JavaScript
        </li>
        <li
          class="skill"
        >
          React
        </li>
      </ul>
    </div>
    <div
      class="education"
    >
      <h2
        class="title"
      >
        Education
      </h2>
      <ul
        class="sub"
      >
        <li>
          Bachelor's Degree
        </li>
      </ul>
    </div>
  </div>
</div>
`;
